### PR TITLE
CHE-2654: Changing Agent REST API (adding 'id' and 'description' fiel…

### PR DESCRIPTION
**Pull request by ibuziuk**
### What does this PR do?

Changes Agent REST API for using 'id' instead of 'name' (adding 'id' and 'description' fields). Updating "Agents" section in the dashboard for using new API
### What issues does this PR fix or reference?

https://github.com/eclipse/che/issues/2654
### Previous behavior

Agent names looked more like internal ids / No description field 
### New behavior

agents

Signed-off-by: Ilya Buziuk ibuziuk@redhat.com

### CODE REVIEW

**Comment by ibuziuk** [comment]
Agent's descriptions might be different (please, advice more suitable options in this case)

File: dashboard/src/app/workspaces/workspace-details/environments/list-agents/list-agents.controller.ts [hunk 1]
```diff
@@ -27,42 +27,48 @@ export class ListAgentsController {
...
-      let isEnabled = this.isEnabled(agent, this.agents);
-      this.agentsList.push({ "name": agent, "isEnabled": isEnabled });
+    this.agentIds = this.cheAgent.getAgents();
+    this.agentIds.forEach(id => {
+
+      this.cheAgent.fetchAgentDetails(id).then(() => {
+        let agent = this.cheAgent.getAgentById(id);
+
+        let isEnabled = this.isEnabled(agent.id, this.agents);
+        this.agentsList.push({
```
Comment (ashumilova) [inline dashboard/src/app/workspaces/workspace-details/environments/list-agents/list-agents.controller.ts]: Here we can do :

```
let agentItem = angular.copy(agent);
agentItem.isEnable = isEnabled;
this.agentsList.push(agentItem);
```

**Comment by ashumilova** [comment]
@tolusha , @ibuziuk can [`List<String> getAgents()`](https://github.com/ibuziuk/che/blob/5fd73693d0a790fc6c21b8eaeeeb53802829dc9d/wsmaster/che-core-api-agent/src/main/java/org/eclipse/che/api/agent/server/AgentRegistryService.java#L117) return list of agent objects? (this will avoid fetching agent's details for each item from the list)

**Comment by ibuziuk** [comment]
@ashumilova @tolusha  this can be done, but probably returning ids instead has been done deliberately for some reason ?
1. [`AgentRegistryImpl.java`](https://github.com/ibuziuk/che/blob/5fd73693d0a790fc6c21b8eaeeeb53802829dc9d/wsmaster/che-core-api-agent/src/main/java/org/eclipse/che/api/agent/server/impl/AgentRegistryImpl.java#L81)
2. [`LocalAgentRegistryImpl.java`](https://github.com/ibuziuk/che/blob/5fd73693d0a790fc6c21b8eaeeeb53802829dc9d/wsmaster/che-core-api-agent/src/main/java/org/eclipse/che/api/agent/server/impl/LocalAgentRegistryImpl.java#L86)

**Comment by tolusha** [comment]
@ibuziuk @ashumilova 
`List<Agents> getAgents()` it is ok for me.

**Comment by ibuziuk** [comment]
@tolusha how the [`AgentRegistryImpl.java`](https://github.com/ibuziuk/che/blob/5fd73693d0a790fc6c21b8eaeeeb53802829dc9d/wsmaster/che-core-api-agent/src/main/java/org/eclipse/che/api/agent/server/impl/AgentRegistryImpl.java#L81) should be refactored in this case?

**Comment by tolusha** [comment]
@ibuziuk 
Let's remove it. It isn't used at all.

**Comment by ibuziuk** [comment]
@tolusha sounds good ;-) 
I will update PR

**Comment by ashumilova** [comment]
Thank's a lot

**Comment by tolusha** [comment]
@ibuziuk Do we need `name` field along with `id` ?

**Comment by ibuziuk** [comment]
@tolusha yes IMO. `"name": "Terminal"` vs. `"id": "org.eclipse.che.terminal"` 
https://github.com/eclipse/che/issues/2654

**Comment by tolusha** [comment]
It  duplicates a bit `description` field then but OK.

**Comment by ibuziuk** [comment]
@tolusha that is why I commented above: 

> Agent's descriptions might be different (please, advice more suitable options in this case)

Agent's name + " support" does not seem to be the right thing. However, we can also leave it this way for now, because changing description is just updating the json file (can be done once the right description will be found"

**Comment by ashumilova** [comment]
+1 on keeping name - it is display human readable name. And some widgets may not contain description - only names as short form.

**Comment by ibuziuk** [comment]
@ashumilova @tolusha PR has been updated

File: wsmaster/che-core-api-agent/src/main/resources/agents/org.eclipse.che.ls.csharp.json [hunk 2]
```diff
@@ -1,5 +1,7 @@
 {
-"name": "org.eclipse.che.ls.csharp",
+"id" : "org.eclipse.che.ls.csharp",
+"name": "C#",
```
Comment (tolusha) [inline wsmaster/che-core-api-agent/src/main/resources/agents/org.eclipse.che.ls.csharp.json]: C# language server

File: wsmaster/che-core-api-agent/src/main/resources/agents/org.eclipse.che.ls.json.json [hunk 3]
```diff
@@ -1,5 +1,7 @@
 {
-"name": "org.eclipse.che.ls.json",
+"id": "org.eclipse.che.ls.json",
+"name": "JSON",
```
Comment (tolusha) [inline wsmaster/che-core-api-agent/src/main/resources/agents/org.eclipse.che.ls.json.json]: JSON language server

File: wsmaster/che-core-api-agent/src/main/resources/agents/org.eclipse.che.ls.php.json [hunk 4]
```diff
@@ -1,5 +1,7 @@
 {
-"name": "org.eclipse.che.ls.php",
+"id": "org.eclipse.che.ls.php",
+"name": "PHP",
```
Comment (tolusha) [inline wsmaster/che-core-api-agent/src/main/resources/agents/org.eclipse.che.ls.php.json]: PHP language server

**Review by tolusha — APPROVED** [review]

**Comment by ibuziuk** [comment]
@tolusha thanks for review - I will update PR later today

**Comment by ibuziuk** [comment]
@tolusha @ashumilova PR has been updated

**Review by ashumilova — APPROVED** [review]

**Comment by ibuziuk** [comment]
PR has been rebased with minor corrections

**Comment by ashumilova** [comment]
@ibuziuk thank you, we are waiting for M6 to release and will merge it.

**Comment by ibuziuk** [comment]
@ashumilova most welcome ;-)

**Comment by slemeur** [comment]
Very nice!

**Comment by ashumilova** [comment]
@ibuziuk can you align PR with master and let's merge it?

**Comment by ibuziuk** [comment]
@ashumilova done

**Pull request merged**